buildkite/resource_team_member.go

```diff
@@ -3,174 +3,208 @@ package buildkite
 import (
 	"context"
 	"fmt"
-
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
-	"github.com/shurcooL/graphql"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-type TeamMemberNode struct {
-	ID   graphql.String
-	Role TeamMemberRole
-	UUID graphql.String
-	Team TeamNode
-	User struct {
-		ID graphql.ID
-	}
+type teamMemberResourceModel struct {
+	Id     types.String `tfsdk:"id"`
+	Uuid   types.String `tfsdk:"uuid"`
+	Role   types.String `tfsdk:"role"`
+	TeamId types.String `tfsdk:"team_id"`
+	UserId types.String `tfsdk:"user_id"`
 }
 
-func resourceTeamMember() *schema.Resource {
-	return &schema.Resource{
-		CreateContext: CreateTeamMember,
-		ReadContext:   ReadTeamMember,
-		UpdateContext: UpdateTeamMember,
-		DeleteContext: DeleteTeamMember,
-		Importer: &schema.ResourceImporter{
-			State: schema.ImportStatePassthrough,
-		},
+type teamMemberResource struct {
+	client *Client
+}
 
-		Schema: map[string]*schema.Schema{
-			"role": &schema.Schema{
-				Required: true,
-				Type:     schema.TypeString,
-				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
-					v := val.(string)
-					switch v {
-					case "MEMBER":
-					case "MAINTAINER":
-						return
-					default:
-						errs = append(errs, fmt.Errorf("%q must be either MEMBER or MAINTAINER, got: %s", key, v))
-						return
-					}
-					return
+func newTeamMemberResource() resource.Resource {
+	return &teamMemberResource{}
+}
+
+func (teamMemberResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team_member"
+}
+
+func (tm *teamMemberResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	tm.client = req.ProviderData.(*Client)
+}
+
+func (teamMemberResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resource_schema.Schema{
+		MarkdownDescription: "A team member resourc allows for the management of team team membership for existing organization users.",
```

Comment:
resource* and double "team"

Reply:
Strikes again - thanks: adjusting in to-come-commit